File: android_mobile/src/main/java/com/alexstyl/specialdates/date/AndroidDateLabelCreator.java
package com.alexstyl.specialdates.date;

import android.content.Context;
import android.text.format.DateUtils;

import com.alexstyl.specialdates.search.DateLabelCreator;

public final class AndroidDateLabelCreator implements DateLabelCreator {

    private final Context context;

    public AndroidDateLabelCreator(Context context) {
        this.context = context;
    }

    @Override
    public String createLabelWithoutYearFor(Date date) {
        int formatFlags = DateUtils.FORMAT_SHOW_DATE | DateUtils.FORMAT_NO_YEAR;
        return DateUtils.formatDateTime(context, date.toMillis(), formatFlags);
    }

    @Override
    public String createLabelWithYearFor(Date date) {
        int formatFlags = DateUtils.FORMAT_SHOW_DATE | DateUtils.FORMAT_SHOW_YEAR;
        return DateUtils.formatDateTime(context, date.toMillis(), formatFlags);
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/date/AndroidDateLabelCreator.kt
package com.alexstyl.specialdates.date

import android.content.Context
import android.text.format.DateUtils

class AndroidDateLabelCreator(private val context: Context) : DateLabelCreator {
    override fun createLabelWithoutYear(date: Date): String {
        val formatFlags = DateUtils.FORMAT_SHOW_DATE or DateUtils.FORMAT_NO_YEAR
        return DateUtils.formatDateTime(context, date.toMillis(), formatFlags)
    }

    override fun createWithYearPreferred(date: Date): String {
        return if (date.hasYear()) {
            val formatFlags = DateUtils.FORMAT_SHOW_DATE or DateUtils.FORMAT_SHOW_YEAR
            DateUtils.formatDateTime(context, date.toMillis(), formatFlags)
        } else {
            val formatFlags = DateUtils.FORMAT_SHOW_DATE or DateUtils.FORMAT_NO_YEAR
            DateUtils.formatDateTime(context, date.toMillis(), formatFlags)
        }
    }
}
